prov/verbs/src/fi_verbs.c

```diff
@@ -815,25 +792,63 @@ static int fi_ibv_init_info(const struct fi_info *hints)
 	default:
 		FI_INFO(&fi_ibv_prov, FI_LOG_CORE, "Unknown transport type");
 		ret = -FI_ENODATA;
-		goto err3;
+		goto err;
 	}
 
 	if (!(fi->domain_attr->name = strdup(ctx->device->name))) {
 		ret = -FI_ENOMEM;
-		goto err3;
+		goto err;
 	}
 
-	verbs_info = fi;
-	rdma_free_devices(ctx_list);
-unlock:
-	pthread_mutex_unlock(&verbs_info_lock);
+	*info = fi;
 	return 0;
-
-err3:
+err:
 	fi_freeinfo(fi);
-err2:
+	return ret;
+}
+
+static int fi_ibv_init_info(const struct fi_info *hints)
```

Comment:
The hints is passed along to fi_ibv_get_info_ctx -> fi_ibv_get_device_attrs -> fi_ibv_get_qp_cap to check if tx_attr->inject_size provided by the user is supported by the QP. The inject size is passed as ibv_qp_init_attr.max_inline_data. I'm not sure if there is another way to check the maximum inject size supported by the device.